src/connector/src/parser/unified/upsert.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::types::DataType;
+
+use super::{Access, ChangeEvent, ChangeEventOperation};
+use crate::parser::unified::AccessError;
+
+/// `UpsertAccess` wraps a key-value message format into an upsert source.
+/// A key accessor and a value accessor are required.
+pub struct UpsertChangeEvent<K, V> {
+    key_accessor: Option<K>,
+    value_accessor: Option<V>,
+    primary_key_column_name: Option<String>,
```

Comment:
I am opposed to this part. any hint on limiting pk to only one column?

Reply:
Maybe the name has caused some misunderstandings, I have rename it `to key_as_column_name`.